### ISSUE TITLE
Incorrect NaN gradient for CUDA broadcast

### ISSUE THREAD

**Issue by marius311** [FluxML/Zygote.jl#942]
Julia 1.6, CUDA v2.6.1, Zygote v0.6.3: 

```julia
using CUDA, Zygote
x = cu([0.])
gradient(α -> sum(@. x ^ α), 1) # NaN
```

This works correctly without the `cu`.

**Comment by DhairyaLGandhi** [comment]
@maleadt to keep him updated

It might be on the CUDA side, so not sure how we are placed to alter it's behavior to suit Julia's defaults

**Comment by marius311** [comment]
Could be wrong, but I had guessed the source of this is the CUDA specific code in broadcast.jl of this repo, hence why I filed it here.

**Comment by marius311** [comment]
This looks to be fixed with Zygote v0.6.9 and CUDA v3.0.2 (haven't tracked down exactly where, if I had to guess its the new broadcasting stuff in CUDA 3 since `pow`, so potentially I did file this in the wrong repo)
